source_gen/test/output_helpers_test.dart

```diff
@@ -49,7 +49,7 @@ void main() {
 
     _testFunction(
       'new Future.error',
-      () => Future.error(ArgumentError('Error message')),
+      () => Future<void>.error(ArgumentError('Error message')),
```

Comment:
[nit] Unless this causes a static error...

```suggestion
      () => Future<Never>.error(ArgumentError('Error message')),
```